crates/refunder/src/refund_service.rs

```diff
@@ -98,63 +93,55 @@ impl RefundService {
     async fn identify_uids_refunding_status_via_web3_calls(
         &self,
         refundable_order_uids: Vec<EthOrderPlacement>,
-    ) -> Result<Vec<OrderUid>> {
+    ) -> Vec<(OrderUid, Vec<CoWSwapEthFlowAddress>)> {
```

Comment:
Given that an `OrderUid` can only belong to a single contract we can ditch the vector. And I would suggest to simply copy the contract type instead of messing around with the address to make the code easier to read. The refunder is doing barely any work anyway so I think we should optimize for readability over performance here.
```suggestion
    ) -> Vec<(OrderUid, CoWSwapEthFlow)> {
```

Reply:
Using contract instead of address is not convenient since PartialEq and Hash are not implemented for CoWSwapEthFlow